### ISSUE TITLE
Issues setting up the default applications directory to ~/Applications

### ISSUE THREAD

**Issue by designt** [Homebrew/homebrew-cask#62666]
After reading the Brew Cask manual on the following page: https://github.com/Homebrew/homebrew-cask/blob/master/USAGE.md

I add the following code to .bash_profile:

export HOMEBREW_CASK_OPTS="--appdir=~/Applications --fontdir=/Library/Fonts"
I source the .bash_profile

~brew --version                                    
Homebrew 2.1.1
Homebrew/homebrew-core (git revision 0124; last commit 2019-04-29)
Homebrew/homebrew-cask (git revision 88ac9; last commit 2019-04-29)
PS: I have my disc with three APFS volumes, system, user and resources. So as you see I moved my user folder using the "Users & Groups" preferences.

Applications keep going to /Applications folder and not to ~/Applications.

It seams to be a bug.

**Comment by vitorgalvao** [comment]
What casks have you tried installing?

**Comment by designt** [comment]
I tried with docker, vagrant and others it seams it´s happening with all casks.

**Comment by designt** [comment]
These are a small example of the casks that é need:
iterm2, textwrangler, SymbolicLinker, virtualbox, docker, vagrant.

**Comment by designt** [comment]
Hi @vitorgalvao,

Thanks for testing this, Can you please confirm if you have the same OSX version and Brew & Cask versions ?

Sorry this is the first time I try to post here, I am copying from iterm and I use "Insert code".
Is this correct ?

This is a new MacOS setup and I´ve check that I still have the previous user folder in Users so I rename the user folder and tried again.

`brew cask install symboliclinker` goes to my user folder in the other volume but iTerm still goes to /Applications folder

` brew cask install iterm2                                                                            
Updating Homebrew...
==> Satisfying dependencies
==> Downloading https://iterm2.com/downloads/stable/iTerm2-3_2_9.zip
Already downloaded: /Volumes/USERS/antila/Library/Caches/Homebrew/downloads/30c1b0d5da5b947edede6ee44d2ae1e8f1a331955d291540f42456bb5836e4c9--iTerm2-3_2_9.zip
==> Verifying SHA-256 checksum for Cask 'iterm2'.
==> Installing Cask iterm2
==> Moving App 'iTerm.app' to '/Applications/iTerm.app'.
 iterm2 was successfully installed!`

`brew cask install xquartz` also go to the boot volume /Applications/Utilities
And also states `==> Package installers may write to any location; options such as --appdir are ignored.`
Does this means that we can not control de packages installations ? 

I am using Zsh shell with Oh-My-Zsh, so should I also add this to their dotfiles ?

# Specify your defaults in this environment variable
export HOMEBREW_CASK_OPTS="--appdir=~/Applications --fontdir=~/Library/Fonts"

Basically I would like to have all my applications in my user folder in another , that´s why I moved my user folder.
I also thought to symlink the /Applications folder but I really do not think if it will work for all apps.
Will Cask update the installed applications if I move them to other folder or volume ?

Thanks a lot.

**Comment by vitorgalvao** [comment]
> Thanks for testing this, Can you please confirm if you have the same OSX version and Brew & Cask versions ?

This feature has been in place for years. It’s not a versioning problem.

> Does this means that we can not control de packages installations ?

Yes.

> I am using Zsh shell with Oh-My-Zsh, so should I also add this to their dotfiles ?

You should add the `export` line to the dotfiles of whatever shell you’re using. If you’re not using Bash, then adding them to bash dotfiles does nothing.

Try this: open a terminal, paste and run the `export` line, and then run `brew cask install iterm2`. It should work. It seems that the issue might stem from an incomprehension of the shell. Unfortunately, this is not the forum to solve such basic questions, and you’ll need to direct them somewhere else. We already get too many reports to deal with regarding HBC itself; we have no bandwidth to debug other tools, and quite frankly your posts are disorganised and hard to read.

I recommend you read the guide for [Github Flavoured Markdown](https://guides.github.com/features/mastering-markdown/) and direct any shell-related questions to a QA website such as [StackOverflow](https://stackoverflow.com/). This is not a Homebrew Cask bug.
